components/member-profile/contribution/index.js

```diff
@@ -0,0 +1,98 @@
+import { useState } from 'react';
+import PropTypes from 'prop-types';
+import classNames from './contribution.module.scss';
+import PRLink from './pr-link/';
+import { timeWas } from '../../../helper-functions/time-was';
+
+const renderPRLinks = (prList) =>
+  prList.map(({ url }, index) => {
+    return <PRLink link={url} key={index} />;
+  });
+
+const Contribution = ({ contribution, fullName, imageLink }) => {
+  const {
+    task: { title, startedOn, endsOn, purpose, featureUrl, status },
+    prList
+  } = contribution;
+  const [showMoreContent, setShowMoreContent] = useState(false);
+
+  const showMoreContentHandler = () => {
+    setShowMoreContent((prevstate) => !prevstate);
+  };
+
+  const showText = showMoreContent ? 'Show less' : 'Show more';
+  const showMoreContentClass = showMoreContent ? classNames.showContent : classNames.hideContent;
+
+  const isTaskAvailable = title ? true : false;
+
+  let completedText = '';
+  let completedDate = '';
+  let featureLiveOnText = '';
+  let featurLiveDate = '';
+
+  if (isTaskAvailable) {
+    if (status === 'Active') {
+      completedText = <span>Estimated Completion: </span>;
+      completedDate = timeWas(startedOn * 1000, false, endsOn * 1000);
+    } else {
+      completedDate = timeWas(endsOn * 1000);
+      completedText = <span>Completed in: </span>;
+      featurLiveDate = timeWas(endsOn * 1000, true);
+      featureLiveOnText = `Feature live on ${featurLiveDate}`;
+    }
+  } else {
+    const createdAt = +new Date(prList[0]['createdAt']);
+    const updatedAt = +new Date(prList[0]['updatedAt']);
+    if (prList[0]['state'] === 'closed') {
+      completedDate = timeWas(createdAt, false, updatedAt);
+      completedText = <span>Completed in: </span>;
+      featurLiveDate = timeWas(updatedAt, true);
+      featureLiveOnText = `Feature live on ${featurLiveDate}`;
+    }
+  }
```

Comment:
Is there a way to refactor this? I am not able to figure it out. Open to any sort of suggestion.